decidim-core/app/mailers/decidim/decidim_devise_mailer.rb

```diff
@@ -1,11 +1,55 @@
 # frozen_string_literal: true
 module Decidim
   # A custom mailer for Devise so we can tweak the invitation instructions for
-  # each role.
+  # each role and use a localised version.
   class DecidimDeviseMailer < Devise::Mailer
-    def invitation_instructions(record, token, opts = {})
-      @token = token
-      devise_mail(record, opts[:invitation_instructions] || :invitation_instructions, opts)
+    # Sends an email with the instructions to confirm an email.
+    #
+    # user  - The User that needs to confirm the email address.
+    # token - The String to be sent as a token to verify the request.
+    # opts  - A Hash with options to send the email (optional).
+    def confirmation_instructions(user, token, opts = {})
+      I18n.with_locale(user.locale || I18n.default_locale) do
```

Comment:
In `decidim.barcelona`, I defined a `with_user` method that received a user and handled this logic, and included it via a module. It also set a `@current_user` variable that was really helpful in layouts and other mailers.

You can then use that helper in `devise_mail` instead of manually in each method. Check it out: https://github.com/AjuntamentdeBarcelona/decidim.barcelona/blob/master/app/mailers/devise_mailer.rb

It also gives you the opportunity to factor out this logic in a more elegant way.